src/routes/documents/[id]-[slug]/+page.svelte

```diff
@@ -95,37 +61,29 @@
     }
   });
 
-  function setMode(e) {
-    const mode = e.target.value;
+  $: {
     const u = new URL($page.url);
-
-    u.searchParams.set("mode", mode);
-
-    // reset hash, keeping page number
-    u.hash = "";
-    if ($currentPage > 1) {
+    // When the mode changes, update the URL param to match.
+    u.searchParams.set("mode", $currentMode);
+    // Reset the hash, keeping page number if we should paginate.
+    if (shouldPaginate($currentMode) && $currentPage > 1) {
       u.hash = pageHashUrl($currentPage);
+    } else {
+      u.hash = "";
     }
-
-    goto(u);
-  }
-
-  // pagination
-  function onHashChange(e: HashChangeEvent) {
-    const { hash } = new URL(e.newURL);
-    $currentPage = pageFromHash(hash);
-    scrollToPage($currentPage);
-
-    const noteId = noteFromHash(hash);
-    if (noteId) {
-      $activeNote = document.notes.find((note) => note.id === noteId);
+    // Only navigate when the mode changes.
+    // Since we're accessing $page or $currentPage, this will run when
+    // those values change, as well. We want to avoid unnecessary runs.
+    if (previousMode !== $currentMode) {
+      goto(u);
```

Comment:
We can't use `goto` on the server, so putting this block in a reactive variable is causing an error.